Sources/DistributedActors/Cluster/ClusterControl.swift

```diff
@@ -155,4 +155,116 @@ public struct ClusterControl {
     }
+
+    /// Wait, within the given duration, until this actor system has joined the node's cluster.
+    ///
+    /// - Parameters
+    ///   - node: The node to be joined by this system.
+    ///   - within: Duration to wait for.
+    ///
+    /// - Returns `Cluster.Member` for the joined node.
+    public func joined(node: UniqueNode, within: Duration) async throws -> Cluster.Member {
+        try await self.waitFor(node, .up, within: within)
+    }
+
+    /// Wait, within the given duration, for this actor system to be a member of all the nodes' respective cluster and have the specified status.
+    ///
+    /// - Parameters
+    ///   - nodes: The nodes to be joined by this system.
+    ///   - status: The expected member status.
+    ///   - within: Duration to wait for.
+    public func waitFor(_ nodes: Set<UniqueNode>, _ status: Cluster.MemberStatus, within: Duration) async throws {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for node in nodes {
+                group.addTask {
+                    _ = try await self.waitFor(node, status, within: within)
+                }
+            }
+            // loop explicitly to propagagte any error that might have been thrown
+            for try await _ in group {}
+        }
+    }
+
+    /// Wait, within the given duration, for this actor system to be a member of all the nodes' respective cluster and have **at least** the specified status.
+    ///
+    /// - Parameters
+    ///   - nodes: The nodes to be joined by this system.
+    ///   - status: The minimum expected member status.
+    ///   - within: Duration to wait for.
+    public func waitFor(_ nodes: Set<UniqueNode>, atLeast atLeastStatus: Cluster.MemberStatus, within: Duration) async throws {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for node in nodes {
+                group.addTask {
+                    _ = try await self.waitFor(node, atLeast: atLeastStatus, within: within)
+                }
+            }
+            // loop explicitly to propagagte any error that might have been thrown
+            for try await _ in group {}
+        }
+    }
+
+    /// Wait, within the given duration, for this actor system to be a member of the node's cluster and have the specified status.
+    ///
+    /// - Parameters
+    ///   - node: The node to be joined by this system.
+    ///   - status: The expected member status.
+    ///   - within: Duration to wait for.
+    ///
+    /// - Returns `Cluster.Member` for the joined node.
+    public func waitFor(_ node: UniqueNode, _ status: Cluster.MemberStatus, within: Duration) async throws -> Cluster.Member {
+        try await self.waitForMembershipEventually(within: within) { membership in
+            guard let foundMember = membership.uniqueMember(node) else {
+                throw Cluster.MembershipError.notFound(node, in: membership)
+            }
+
```

Comment:
Oh sorry to bother you but I thought of another case...

If a node was removed BEFORE, we'll have a tombstone for it inside the `    private var _associationTombstones: [UniqueNode: Association.Tombstone]` protected by `_associationsLock`. If it is in there, we should return that the member was `.removed`

You can "make up" a member for a removed node by creating a `Member(` from the unique node and put `.unreachable` and `.removed` in there 👍 

This could be in a separate PR  👍

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/972/commits/5178fa44108777eb2c921ac5fd9279f048ee6bab

Reply:
Looking good!